src/3d/qgscameracontroller.cpp

```diff
@@ -1057,15 +1058,19 @@ void QgsCameraController::setMouseParameters( const MouseOperation &newOperation
     return;
   }
 
+  if ( newOperation == MouseOperation::None )
+  {
+    mClickPoint = QPoint();
+  }
   // click point and rotation angles are updated if:
   // - it has never been computed
   // - the current and new operations are both rotation and translation
   // Indeed, if the sequence such as rotation -> zoom -> rotation updating mClickPoint on
   // the click point does not need to be updated because the relative mouse position is kept
   // during a zoom operation
-  if ( mClickPoint.isNull() ||
-       ( ( newOperation == MouseOperation::Rotation || newOperation == MouseOperation::Translation ) &&
-         ( mCurrentOperation == MouseOperation::Rotation ||  mCurrentOperation == MouseOperation::Translation ) ) )
+  else if ( mClickPoint.isNull() ||
+            ( std::find( mTranslateOrRotate.begin(), mTranslateOrRotate.end(), newOperation ) != std::end( mTranslateOrRotate ) &&
```

Comment:
can you add a small function to do this search? It could make this condition more readable :wink:

Reply:
Done.